servers/src/main/java/tachyon/worker/block/BlockMetadataManagerView.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.HashSet;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.master.BlockInfo;
+import tachyon.worker.block.meta.BlockMeta;
+import tachyon.worker.block.meta.StorageTier;
+import tachyon.worker.block.meta.StorageTierView;
+
+/**
+ * This class exposes a narrower view of {@link BlockMetadataManager} to Evictors and Allocators.
+ */
+public class BlockMetadataManagerView {
+
+  /** The BlockMetadataManager this view is derived from */
+  private final BlockMetadataManager mMetadataManager;
+  /** A list of StorageTierView, derived from StorageTiers from the BlockMetadataManager */
+  private List<StorageTierView> mTierViews = new ArrayList<StorageTierView>();
+  /** A list of pinned inodes */
+  private final Set<Integer> mPinnedInodes = new HashSet<Integer>();
+  /** A list of blocks that are currently being read */
+  private final Set<Long> mLockedBlocks = new HashSet<Long>();
+
+  /**
+   * Create a BlockMatadataManagerView.
+   * Now we always creating a new view before freespace.
+   * TODO: incrementally update the view
+   *
+   * @param manager
+   * @param pinnedBlocks
+   * @param lockedBlocks
+   * @throws IOException
+   */
+  public BlockMetadataManagerView(BlockMetadataManager manager, Set<Integer> pinnedInodes,
+      Set<Long> lockedBlocks) throws IOException {
+    mMetadataManager = Preconditions.checkNotNull(manager);
+    mPinnedInodes.addAll(Preconditions.checkNotNull(pinnedInodes));
+    mLockedBlocks.addAll(Preconditions.checkNotNull(lockedBlocks));
```

Comment:
Instead of initializing the sets in the declaration, would it be easier to just set `mPinnedInodes = Preconditions.checkNotNull(pinnedInodes)` and something similar for locked blocks?